synapse/replication/tcp/client.py

```diff
@@ -226,7 +230,9 @@ async def on_rdata(
         # Notify any waiting deferreds. The list is ordered by position so we
         # just iterate through the list until we reach a position that is
         # greater than the received row position.
-        waiting_list = self._streams_to_waiters.get((stream_name, instance_name), [])
+        waiting_list: MutableSequence[
+            Tuple[int, Deferred]
+        ] = self._streams_to_waiters.get((stream_name, instance_name), [])
```

Comment:
Why a `MutableSequence`? Because it might be a `SortedList` or a `List`? Should we be defaulting to a `SortedList` here or are we just abusing that we *know* it is sorted?

Reply:
Yeah, it made me sad to instansiate a new `SortedList`, but actually we could bypass that by just returning early...

Reply:
I think that's clearer! And always better to just skip code if possible...

Reply:
Indeed!